.github/workflows/generate_matrix_page.yml

```
@@ -43,31 +43,34 @@ jobs:
+          git merge origin/main --strategy-option=theirs --allow-unrelated-histories --no-ff -m "Merge main into PR branch"  # Merge main into the PR branch
 
-      # Step 7: Commit and push updated JSON to the PR branch
+      # Step 7: Set git user identity for commits
+      - name: Set Git identity
+        run: |
+          git config user.name "${{ github.actor }}"
+          git config user.email "${{ github.actor }}@users.noreply.github.com"
+
+      # Step 8: Commit and push updated JSON to the PR branch
       - name: Commit and push updated JSON data to the PR branch
         run: |
```

actionlint (1.7.4)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)